File: 프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string my_string, string overwrite_string, int s) {
    string answer = my_string;
    for (int idx = 0; idx < overwrite_string.size(); idx++)
    {
        // if (idx >= s && idx <= overwrite_string.size())
            answer[s+idx] = overwrite_string[idx];
        // else
            // answer[idx] = my_string[idx];
    }
    return answer;
}

File: 프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.py
def solution(my_string, overwrite_string, s):
    answer = my_string
    
    end_idx = s + len(overwrite_string)
    
    answer = my_string[:s] + overwrite_string + my_string[end_idx:]
    # answer_list = list(my_string)
    # for idx, ch in enumerate(overwrite_string):
    #     answer_list[s + idx] = overwrite_string[idx]
    # answer = "".join(answer_list)
    return answer